projects/igniteui-angular/src/lib/checkbox/checkbox.component.ts

```diff
@@ -493,28 +494,40 @@ export class IgxCheckboxComponent implements EditorProvider, AfterViewInit, Cont
      */
     private _required = false;
     private elRef = inject(ElementRef);
-    private _theme$ = new Subject<IgxTheme>();
-    private _subscription: Subscription;
     private destroyRef = inject(DestroyRef);
 
     constructor(
         protected cdr: ChangeDetectorRef,
         protected renderer: Renderer2,
-        protected themeService: ThemeService,
+        @Inject(THEME_TOKEN)
+        protected themeToken: ThemeToken,
         @Optional() @Self() public ngControl: NgControl,
     ) {
         if (this.ngControl !== null) {
             this.ngControl.valueAccessor = this;
         }
 
-        this.theme = this.themeService.globalTheme;
+        this.theme = this.themeToken.theme;
 
-        this._subscription = this._theme$.asObservable().subscribe(value => {
-            this.theme = value as IgxTheme;
-            this.cdr.detectChanges();
+        const { unsubscribe } = this.themeToken.onChange((theme) => {
+            if (this.theme !== theme) {
+                this.theme = theme;
+                this.cdr.detectChanges();
+            }
         });
 
-        this.destroyRef.onDestroy(() => this._subscription.unsubscribe());
+        this.destroyRef.onDestroy(() => unsubscribe);
+    }
+
+    private setComponentTheme() {
+        if(!this.themeToken.preferToken) {
+            const theme = getComponentTheme(this.elRef.nativeElement);
+
+            if (theme && theme !== this.theme) {
+                this.themeToken.set(theme);
```

Comment:
Sorry, missed this one - shouldn't this be `this.theme = theme` instead of setting it on the token which the component doesn't own and is in this case ignoring too?

Reply:
You are correct. This shouldn't modify the global token theme.